File: android/src/main/java/com/mycompany/plugins/example/VNPaySDKIntegratedPlugin.java
package com.mycompany.plugins.example;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
//import com.getcapacitor.annotation.CapacitorPlugin;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.NativePlugin;


//@CapacitorPlugin(name = "VNPaySDKIntegrated")
@NativePlugin(
  requestCodes = {123123}
)
public class VNPaySDKIntegratedPlugin extends Plugin {

    private VNPaySDKIntegrated implementation = new VNPaySDKIntegrated();

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);
    }
}

File: ios/Plugin/VNPaySDKIntegratedPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(VNPaySDKIntegratedPlugin)
public class VNPaySDKIntegratedPlugin: CAPPlugin {
    private let implementation = VNPaySDKIntegrated()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        DispatchQueue.main.async {
            ViewController().openSDK(self: self.bridge.viewController)
        }
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
}
